src/Common/tests/System.Net/Sockets/SocketTestServerAsync.cs

```diff
@@ -168,23 +172,39 @@ private void ProcessAccept(SocketAsyncEventArgs e)
         {
             if (e.SocketError != SocketError.Success)
             {
-                return;
-            }
+                _log.WriteLine(this.GetHashCode() + " ProcessAccept(m_numConnectedSockets={0}): {1}", m_numConnectedSockets, e.SocketError);
 
-            Interlocked.Increment(ref m_numConnectedSockets);
-            _log.WriteLine(this.GetHashCode() + " ProcessAccept(m_numConnectedSockets={0})", m_numConnectedSockets);
-
-            // Get the socket for the accepted client connection and put it into the ReadEventArg object user token.
-            SocketAsyncEventArgs readEventArgs = m_readWritePool.Pop();
+                // If possible, retry the accept after a short wait.
+                if (e.SocketError == SocketError.OperationAborted || e.SocketError == SocketError.Interrupted)
+                {
+                    return;
+                }
 
-            ((AsyncUserToken)readEventArgs.UserToken).Socket = e.AcceptSocket;
+                if (Interlocked.Decrement(ref m_acceptRetryCount) <= 0)
+                {
+                    Assert.True(false, "accept retry limit exceeded.");
+                    return;
+                }
 
-            // As soon as the client is connected, post a receive to the connection 
-            bool willRaiseEvent = e.AcceptSocket.ReceiveAsync(readEventArgs);
-            if (!willRaiseEvent)
+                Task.Delay(500).Wait();
```

Comment:
The error I was seeing on OS X was "too many open sockets," so I suspect that the per-process limit on the number of sockets is rather low on the same. Delaying and retrying gives the test just enough time for some of those sockets to close, and the test can continue. It might be better to find a more sophisticated solution in the future.